File: php/v3/examples/BulkListAssignedTargetingOptions.php
<?php
/*
 * Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// Require the base class.
require_once __DIR__ . "/../BaseExample.php";

// Require the necessary utility class.
require_once __DIR__ . "/Utils/PrintUtils.php";

/**
 * This example lists all of the targeting options assigned to a line item
 * across targeting types, optionally filtered by a given filter expression.
 */
class BulkListAssignedTargetingOptions extends BaseExample
{
    /**
     * (non-PHPdoc)
     * @see BaseExample::getInputParameters()
     */
    protected function getInputParameters(): array
    {
        return array(
            array(
                'name' => 'advertiser_id',
                'display' => 'Advertiser ID',
                'required' => true
            ),
            array(
                'name' => 'line_item_ids',
                'display' => 'Line Item IDs (comma-separated)',
                'required' => true
            ),
            array(
                'name' => 'filter',
                'display' => 'Filter',
                'required' => false
            )
        );
    }

    /**
     * (non-PHPdoc)
     * @see BaseExample::run()
     */
    public function run()
    {
        $values = $this->formValues;
        $advertiserId = $values['advertiser_id'];
        $lineItemIds = explode(",", $values['line_item_ids']);
        $filter = $values['filter'];

        $retrievedLineItemTargeting = array();

        $response = null;
        $nextPageToken = null;

        do {
            $optParams = array(
                'lineItemIds' => $lineItemIds,
                'filter' => $filter,
                'pageToken' => $nextPageToken
            );

            // Call the API, getting all the assigned targeting options for the
            // identified line item.
            try {
                $response = $this
                    ->service
                    ->advertisers_lineItems
                    ->bulkListAssignedTargetingOptions(
                        $advertiserId,
                        $optParams
                    );
            } catch (\Exception $e) {
                $this->renderError($e);
                return;
            }

            if (!empty($response->getLineItemAssignedTargetingOptions())) {
                $retrievedLineItemTargeting = array_merge(
                    $retrievedLineItemTargeting,
                    $response->getLineItemAssignedTargetingOptions()
                );
            }

            // Update the next page token.
            $nextPageToken = $response->getNextPageToken();
        } while (
            !empty($response->getLineItemAssignedTargetingOptions())
            && !empty($nextPageToken)
        );

        // Print information returned by the bulk edit request.
        PrintUtils::printBulkListTargetingResponse(
            $lineItemIds,
            $retrievedLineItemTargeting
        );
    }

    /**
     * (non-PHPdoc)
     * @see BaseExample::getName()
     */
    public static function getName(): string
    {
        return 'Bulk List Assigned Targeting Options';
    }
}

File: java/v4/src/main/java/com/google/displayvideo/api/samples/BulkListAssignedTargetingOptions.java
// Copyright 2023 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package com.google.displayvideo.api.samples;

import com.beust.jcommander.Parameter;
import com.google.api.client.util.Strings;
import com.google.api.services.displayvideo.v4.DisplayVideo;
import com.google.api.services.displayvideo.v4.DisplayVideo.Advertisers.LineItems;
import com.google.api.services.displayvideo.v4.model.BulkListAssignedTargetingOptionsResponse;
import com.google.api.services.displayvideo.v4.model.LineItemAssignedTargetingOption;
import com.google.displayvideo.api.samples.utils.ArgumentNames;
import com.google.displayvideo.api.samples.utils.CodeSampleParams;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

/**
 * This example lists all of the targeting options assigned to a line item across targeting types.
 */
public class BulkListAssignedTargetingOptions {

  private static class BulkListAssignedTargetingOptionsParams extends CodeSampleParams {

    @Parameter(
        names = ArgumentNames.ADVERTISER_ID,
        description =
            "The ID of the parent advertiser of the line items whose targeting is being retrieved.",
        required = true)
    public Long advertiserId;

    @Parameter(
        names = ArgumentNames.LINE_ITEM_IDS,
        description = "The IDs of the line items whose targeting is being retrieved.",
        required = true)
    public List<Long> lineItemIds;

    @Parameter(
        names = ArgumentNames.FILTER,
        description = "The filter expression by which to filter the list results.")
    public String filter;
  }

  public static void main(String[] args) throws Exception {
    BulkListAssignedTargetingOptionsParams params = new BulkListAssignedTargetingOptionsParams();
    if (!params.parseArguments(args)) {

      // Either pass the required parameters for this example on the command line, or insert them
      // into the code here. See the parameter class definition above for descriptions.
      params.advertiserId = Long.valueOf("INSERT_ADVERTISER_ID_HERE");

      // Convert string list of line item ID values to Long values.
      List<String> lineItemIdStrings = Arrays.asList("INSERT_LINE_ITEM_IDS_HERE".split(","));
      params.lineItemIds =
          lineItemIdStrings.stream()
              .filter(idStr -> !idStr.isEmpty())
              .map(Long::valueOf)
              .collect(Collectors.toList());

      // If no filter is needed, please leave filter empty.
      params.filter = "INSERT_FILTER_HERE";
    }

    DisplayVideo service =
        DisplayVideoFactory.getInstance(
            params.clientSecretsFile,
            params.useServiceAccount,
            params.serviceAccountKeyFile,
            params.additionalScopes);

    runExample(service, params.advertiserId, params.lineItemIds, params.filter);
  }

  public static void runExample(
      DisplayVideo service, long advertiserId, List<Long> lineItemIds, String filter)
      throws Exception {

    // Configure the bulk list request and set filter.
    LineItems.BulkListAssignedTargetingOptions request =
        service
            .advertisers()
            .lineItems()
            .bulkListAssignedTargetingOptions(advertiserId)
            .setLineItemIds(lineItemIds)
            .setFilter(filter);

    // Create the response and nextPageToken variables.
    BulkListAssignedTargetingOptionsResponse response;
    String nextPageToken = null;

    do {
      // Set page token and execute the list request.
      response = request.setPageToken(nextPageToken).execute();

      // Check if response is empty.
      if (response.isEmpty()) {
        System.out.print("Bulk list request returned no assigned targeting options");
        break;
      }

      // Iterate over retrieved assigned targeting options.
      for (LineItemAssignedTargetingOption lineItemAssignedOption :
          response.getLineItemAssignedTargetingOptions()) {
        System.out.printf(
            "Assigned targeting option %s found%n",
            lineItemAssignedOption.getAssignedTargetingOption().getName());
      }

      // Update the next page token.
      nextPageToken = response.getNextPageToken();
    } while (!Strings.isNullOrEmpty(nextPageToken));
  }
}
